packages/block-editor/src/components/inserter-listbox/index.js

```diff
@@ -1,26 +1,30 @@
 /**
  * WordPress dependencies
  */
-import { __unstableUseCompositeState as useCompositeState } from '@wordpress/components';
+import { privateApis as componentsPrivateApis } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import InserterListboxContext from './context';
+import { unlock } from '../../lock-unlock';
 
 export { default as InserterListboxGroup } from './group';
 export { default as InserterListboxRow } from './row';
 export { default as InserterListboxItem } from './item';
 
+const { CompositeV2: Composite, useCompositeStoreV2: useCompositeStore } =
+	unlock( componentsPrivateApis );
+
 function InserterListbox( { children } ) {
-	const compositeState = useCompositeState( {
-		shift: true,
-		wrap: 'horizontal',
+	const store = useCompositeStore( {
+		focusShift: true,
+		focusWrap: 'horizontal',
 	} );
+
 	return (
-		<InserterListboxContext.Provider value={ compositeState }>
+		<Composite store={ store } render={ <></> }>
```

Comment:
What's the `render={ <></> }` for?

Reply:
This is a weird one, and I'm open to input, but originally nothing was rendered by `InserterListbox` – it was purely a context provider. So in order to not add any new elements, we're rendering a fragment.

In hindsight, it would probably be better to instead export `CompositeProvider` from `Composite` and use that here instead.

Reply:
I had a quick look, and even with a `CompositeProvider`, the `Composite` wrapper is still required to actually make things work. So we'd still need this `render={ <></> }` bit regardless, if we want to keep the DOM as before. I'll leave as is for now.